### PR TITLE
Grpc client interceptors

### PR DESCRIPTION
**Pull request by benjigoldberg**
* Adds default GRPC Client Interceptors
* Adds "recovery" handler for panics in the server (eg translate server panics into grpc errors.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,12 +11,12 @@ require (
 	github.com/go-sql-driver/mysql v1.4.1
 	github.com/gorilla/mux v1.7.1
 	github.com/grpc-ecosystem/go-grpc-middleware v1.1.0
-	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
+	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.1-0.20191002090509-6af20e3a5340
```
Comment (briancharous) [inline go.mod]: Intentional?

Comment (benjigoldberg) [reply go.mod]: Yep, they havent published a new version in a long time, and I want to use the latest features

**Pull request merged**